android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/worker/MeasureReportWorker.kt

```diff
@@ -0,0 +1,174 @@
+        }
+      }
+    }
+
+    return Result.success()
+  }
+
+  private fun lastDayOfMonth(date: Date?): Boolean = date?.before(today().lastDayOfMonth()) == true
+
+  private suspend fun evaluatePopulationMeasure(
+    measureUrl: String,
+    startDateFormatted: String,
+    endDateFormatted: String
+  ) {
+    withContext(dispatcherProvider.io()) {
+      fhirEngine.loadCqlLibraryBundle(fhirOperator, measureUrl)
+    }
+    val measureReport: MeasureReport? =
+      withContext(dispatcherProvider.io()) {
+        try {
+          fhirOperator.evaluateMeasure(
+            measureUrl = measureUrl,
+            start = startDateFormatted,
+            end = endDateFormatted,
+            reportType = MeasureReportViewModel.POPULATION,
+            subject = null,
+            practitioner = null
+            /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
+            practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
+            lastReceivedOn = null // Non-null value not supported yet
+          )
+        } catch (exception: IllegalArgumentException) {
+          Timber.e(exception)
+          null
+        }
+      }
+    if (measureReport != null) {
+      val result =
+        alreadyGeneratedMeasureReports(fhirEngine, startDateFormatted, endDateFormatted, measureUrl)
+      if (result.isNotEmpty()) defaultRepository.delete(result.last())
+      defaultRepository.addOrUpdate(resource = measureReport)
+    }
+  }
+
+  /** @return list of months within current date and starting date of campaign */
+  fun getMonthRangeList(): List<Date> {
+    Timber.w("appConfig$applicationConfiguration")
+    val yearMonths = mutableListOf<Date>()
+    val endDate = Calendar.getInstance().time.formatDate(SDF_YYYY_MM_DD).parseDate(SDF_YYYY_MM_DD)
+    var lastDate = endDate?.firstDayOfMonth()
+    while (lastDate?.after(applicationConfiguration.registerDate.parseDate(SDF_YYYY_MM_DD)) ==
+      true) {
+      yearMonths.add(lastDate)
+      lastDate = lastDate.plusMonths(-1)
+    }
+    return yearMonths.toList()
+  }
+  companion object {
+    private const val WORK_ID = "fhirMeasureReportWorker"
+
+    fun scheduleMeasureReportWorker(workManager: WorkManager, hours: Int, minutes: Int) {
+      // trigger at 8:30am
+      val alarmTime = LocalTime.of(hours, minutes)
+      var now = LocalDateTime.now().truncatedTo(ChronoUnit.MINUTES)
+      val nowTime = now.toLocalTime()
+      // if same time, schedule for next day as well
+      // if today's time had passed, schedule for next day
+      if (nowTime == alarmTime || nowTime.isAfter(alarmTime)) {
+        now = now.plusDays(1)
+      }
+      now =
+        now
+          .withHour(alarmTime.hour)
+          .withMinute(alarmTime.minute) // .withSecond(alarmTime.second).withNano(alarmTime.nano)
+      val duration = Duration.between(LocalDateTime.now(), now)
+
+      Timber.d("runAt=${duration.seconds}s")
```

Comment:
Can we improve the log message here by indicating what should be running at the mentioned duraiton?

Reply:
no need of this log, Removing it